rust/tool/src/generation.rs

```diff
@@ -96,7 +95,7 @@ impl fmt::Display for Generation {
 }
 
 fn read_build_time(path: &Path) -> Result<Date> {
-    let build_time = time::OffsetDateTime::from_unix_timestamp(fs::metadata(path)?.mtime())?.date();
+    let build_time = time::OffsetDateTime::from(fs::metadata(path)?.created().unwrap()).date();
```

Comment:
Is it necessary to unwrap here instead of propagating the error?

Reply:
Thanks for the feedback.
I'm a rust newbie and didn't know the difference between unwrap and propagating at the time.

you're right. there's no need to unwrap there. I'll update the PR.

Reply:
Kudos that you gave it a shot! I would wait a bit for @nikstur to look at this.